src/Sylius/Bundle/CoreBundle/PaymentRequest/Processor/Payum/RequestProcessor.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\PaymentRequest\Processor\Payum;
+
+use Payum\Core\Payum;
+use Payum\Core\Security\TokenAggregateInterface;
+use Sylius\Bundle\CoreBundle\PaymentRequest\Payum\PaymentRequestContextInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+
+final class RequestProcessor implements RequestProcessorInterface
+{
+    public function __construct(
+        private Payum $payum,
+        private PaymentRequestContextInterface $payumApiContext,
+    ) {
+    }
+
+    public function process(
+        PaymentRequestInterface $paymentRequest,
+        mixed $request,
```

Comment:
Don't we really know what kind of request we're expecting?

Something like that won't work?

```suggestion
        PaymentRequestInterface|SomeOtherRequestInterface|thirdRequestInterface $request,
```

Reply:
I wanted to set array here instead of mixed since this field must be set by javascript. Right now it could be :

- What we can call a `PayumHttpRequest` with specific data, being basically a clone of an HTTP request.
- Any kind of data when you are building your own payment gateway.

How can we implement this flexibility of input data ?